btsieve/src/bitcoin/mod.rs

```diff
@@ -92,13 +99,36 @@ where
         }
         missing_block_futures = new_missing_block_futures;
 
+        if let Some(block) = oldest_block.clone() {
+            if block.header.time >= reference_timestamp {
```

Comment:
🔧 I find this piece of code quite hacky. Can't we have 2 different logic depending on whether `reference_timestamp` is `Some(..)` or `None`?
Now I have to think "ok, if there is no timestamp provided then this condition is always false..." too much mental load.

```suggestion
		if let Some(reference_timestamp) = reference_timestamp {
            if block.header.time >= reference_timestamp {
```